### PR TITLE
libjpeg-turbo: Fix compiler checks

### PR DESCRIPTION
**Pull request by xclaesse**
This is a translation from CMake's check_c_source_compiles() that
compiles and links the code.

Fixes: #111

### CODE REVIEW

File: subprojects/glib.wrap [hunk 1]
```diff
@@ -1,9 +1,6 @@
-[wrap-file]
-directory = glib-2.68.1
-
-source_url = https://download.gnome.org/sources/glib/2.68/glib-2.68.1.tar.xz
-source_filename = glib-2.68.1.tar.xz
-source_hash = 241654b96bd36b88aaa12814efc4843b578e55d47440103727959ac346944333
+[wrap-git]
+url = https://gitlab.gnome.org/gnome/glib.git
+revision = main
```
Comment (tp-m) [inline subprojects/glib.wrap:3]: Is this meant to be part of this PR?

Comment (xclaesse) [reply subprojects/glib.wrap]: Oops, sorry, leftover from some unrelated tests I did. `git add -p` followed by `git commit -a`....

Dropped that now.

**Comment by eli-schwartz** [comment]
This should just use has_function, following the intent of the check, rather than taking an overly literal translation of the cmake code.

**Comment by xclaesse** [comment]
> This should just use has_function, following the intent of the check, rather than taking an overly literal translation of the cmake code.

Ok, changed that, as long as someone can test it for me... :D

**Comment by jpakkane** [comment]
Since no-one volunteered to test, let's merge.

**Pull request merged**